src/main/vcp_hal/usbd_cdc_interface.c

```diff
@@ -241,35 +246,42 @@ static int8_t CDC_Itf_Control (uint8_t cmd, uint8_t* pbuf, uint16_t length)
   */
 void HAL_TIM_PeriodElapsedCallback(TIM_HandleTypeDef *htim)
 {
-    if (htim->Instance != TIMusb) return;
+    if (htim->Instance != TIMusb) {
+        return;
+    }
 
-    uint32_t buffptr;
     uint32_t buffsize;
+    static uint32_t lastBuffsize = 0;
```

Comment:
BTW: storing 'uncommitted' index instead of `lastBuffsize` is IMO slightly cleaner and more robust